RecoPPS/Local/plugins/DiamondSampicDigiProducer.cc

```diff
@@ -0,0 +1,185 @@
+// -*- C++ -*-
+//
+// Package:    SampicDigi/DiamondSampicDigiProducer
+// Class:      DiamondSampicDigiProducer
+//
+/**\class DiamondSampicDigiProducer DiamondSampicDigiProducer.cc SampicDigi/DiamondSampicDigiProducer/plugins/DiamondSampicDigiProducer.cc
+
+ Description: This plugin takes testbeam data as input and converts them to TotemTimingDigi which could be then passed to reco
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Christopher Misan (krzysztof.misan@cern.ch)
+//         Created:  Sun, 07 Mar 2021 14:42:52 GMT
+//
+//
+
+// system include files
+#include <memory>
+#include <string>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/CTPPSDigi/interface/TotemTimingDigi.h"
+#include "DataFormats/CTPPSDetId/interface/TotemTimingDetId.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+
+#include "TFile.h"
+#include "TTree.h"
+#include "TChain.h"
+
+const int MAX_SAMPIC_CHANNELS = 32;
+const int SAMPIC_SAMPLES = 64;
+
+//
+// class declaration
+//
+
+class DiamondSampicDigiProducer : public edm::stream::EDProducer<> {
```

Comment:
I'm wondering if this shouldn't be `edm::one::EDProducer` instead, given that it relies on TChain IO. Did you check that the current implementation is thread-safe?

Reply:
This module is not to be run in central reconstruction, it's sole purpose is to test the reco changes. Nevertheless, after checking, it seems to be thread-safe.

Reply:
Perhaps this can be moved to /test in this case, if it is used only for testing